### PR TITLE
[WIP] Lua Formatter in Lua

### PR DESCRIPTION
**Pull request by rockerBOO**
This is actually an experimental pull.

Using the added lua-format.lua file, we can format lua files. Maybe we can massage it to work how we want?

I formatted `lua/telescope/pickers.lua` with it as an example.

My workflow right now.
```
 lua lua/lua-format.lua < lua/telescope/pickers.lua > lua/telescope/pickers.lua.out
cp lua/telescope/pickers.lua.out lua/telescope/pickers.lua
```

### CODE REVIEW

**Comment by tjdevries** [comment]
Ah, it makes everything have semi-colons :'(

Let's not do that. haha

Very cool idea though -- perhaps we should make a new repo in nvim-lua for the formatter after this. And then make it easy to run on lua files for nvim or something (with configuration options).

If you have questions about LPEG, let me know -- I've done a decent amount of stuff w/ it actually.

**Comment by rockerBOO** [comment]
Ok how about this version. I just removed all the semicolon logic and it seemed to work. Not sure how to make it configurable at all.

File: lua/telescope/pickers.lua [hunk 1]
```diff
@@ -371,41 +376,48 @@ function Picker:find()
...
 
   vim.cmd [[startinsert!]]
 end
 
 function Picker:hide_preview()
-  -- 1. Hide the window (and border)
-  -- 2. Resize prompt & results windows accordingly
-end
-
+ -- 1. Hide the window (and border)
```
Comment (rockerBOO) [inline lua/telescope/pickers.lua:418]: Not indented properly here.

File: lua/telescope/pickers.lua [hunk 2]
```diff
@@ -301,68 +309,65 @@ function Picker:find()
...
+      on_detach = function (...)
+ -- print("DETACH:", ...)
+               end
+    })
 
   -- TODO: Use WinLeave as well?
-  local on_buf_leave = string.format(
-    [[  autocmd BufLeave <buffer> ++nested ++once :lua require('telescope.pickers').on_close_prompt(%s)]],
-    prompt_bufnr)
+  local on_buf_leave = string.format([[  autocmd BufLeave <buffer> ++nested ++once :lua require('telescope.pickers').on_close_prompt(%s)]], prompt_bufnr)
```
Comment (rockerBOO) [inline lua/telescope/pickers.lua:366]: Long line, kept it all on 1 line.

File: lua/telescope/pickers.lua [hunk 3]
```diff
@@ -301,68 +309,65 @@ function Picker:find()
...
-    on_lines = vim.schedule_wrap(on_lines),
-
-    on_detach = function(...)
-      -- print("DETACH:", ...)
-    end,
-  })
+      on_lines = vim.schedule_wrap(on_lines),
 
+      on_detach = function (...)
+ -- print("DETACH:", ...)
```
Comment (rockerBOO) [inline lua/telescope/pickers.lua:361]: Comment not indented properly.

File: lua/telescope/pickers.lua [hunk 4]
```diff
@@ -229,51 +236,52 @@ function Picker:find()
 
   local selection_strategy = self.selection_strategy or 'reset'
 
-  local on_lines = function(_, _, _, first_line, last_line)
+  local on_lines = function (_, _, _, first_line, last_line)
```
Comment (rockerBOO) [inline lua/telescope/pickers.lua:239]: Additional space on anonymous functions

File: lua/telescope/pickers.lua [hunk 5]
```diff
@@ -416,7 +428,7 @@ function Picker:close_windows(status)
   local results_border_win = status.results_border_win
   local preview_border_win = status.preview_border_win
 
-  local function del_win(name, win_id, force)
+  local function del_win (name, win_id, force)
```
Comment (rockerBOO) [inline lua/telescope/pickers.lua:431]: local function named? adding a space

**Comment by tjdevries** [comment]
I think we should probably just drop these PRs? If we wanted to do this, I think doing it in plenary would be a better place for it.

**Comment by rockerBOO** [comment]
Yeah I can remake them on plenary.

**Pull request closed without merging**